src/blueprint.zig

```diff
@@ -335,6 +337,116 @@ pub const Pattern = struct {
 	}
 };
 
+pub const Mask = struct {
+	entries: ListUnmanaged(Entry),
+
+	pub const separator = ',';
+	pub const inverse = '!';
+	pub const tag = '$';
+	pub const property = '@';
+
+	const Entry = struct {
+		inner: Inner,
+		isInverse: bool,
+
+		const Inner = union(enum) {
+			block: struct {typ: u16, data: ?u16},
+			blockTag: Tag,
+			property: Property,
+
+			const Property = enum {transparent, collide, solid, selectable, degradable, viewThrough, allowOres, isEntity};
+
+			fn initFromString(specifier: []const u8) !Inner {
+				switch(specifier[0]) {
+					tag => {
+						const blockTag = specifier[1..];
+						if(blockTag.len == 0) return error.MaskSyntaxError;
+						return .{.blockTag = Tag.find(blockTag)};
+					},
+					property => {
+						const propertyName = specifier[1..];
+						const propertyValue = std.meta.stringToEnum(Property, propertyName) orelse return error.MaskSyntaxError;
+						return .{.property = propertyValue};
+					},
+					else => {
+						const block = try main.blocks.tryParseBlock(specifier);
+						return .{.block = .{.typ = block.typ, .data = block.data}};
+					},
+				}
+			}
+
+			fn match(self: Inner, block: Block) bool {
+				return switch(self) {
+					.block => block.typ == self.block.typ and (self.block.data == null or block.data == self.block.data),
+					.blockTag => |desired| {
+						for(block.blockTags()) |current| {
+							if(desired == current) return true;
+						}
+						return false;
+					},
+					.property => |prop| return switch(prop) {
+						.transparent => block.transparent(),
+						.collide => block.collide(),
+						.solid => block.solid(),
+						.selectable => block.selectable(),
+						.degradable => block.degradable(),
+						.viewThrough => block.viewThrough(),
+						.allowOres => block.allowOres(),
+						.isEntity => block.entityDataClass() != null,
```

Comment:
This is problematic, now we have a parameter name, that's different to the block parameter name, and can only be used for masks. If someone asks what parameters you can check in masks, do you want to tell them
> "all block parameters hat are bools"

or do you want to tell them
> "all block parameters that are bools, except maybe we forgot to add some, oh and also there is this is one special case `isEntity` which is not a block property, instead it checks if the entityDataClass exists"

If we need any non-bool parameters (and seriously, why would anyone want to replace all blocks with block entities?), then we should probably implement a more generic comparing mechanism.